src/sql/src/plan.rs

```diff
@@ -571,17 +573,17 @@ impl QueryWhen {
         }
     }
     /// Returns whether the candidate must be advanced to the upper.
-    pub fn advance_to_upper(&self, uses_tables: bool) -> bool {
+    pub fn advance_to_upper(&self) -> bool {
         match self {
-            QueryWhen::Immediately | QueryWhen::AtLeastTimestamp(_) => !uses_tables,
+            QueryWhen::Immediately | QueryWhen::AtLeastTimestamp(_) => true,
```

Comment:
Got it, that makes sense. I had not read the relevant docs.

Reply:
IIRC we introduce "AtLeastTimestamp" to be a more robust "AtTImestamp", rather than as a signal for freshness. I'd be up for this being `false` under the spirit of "at least" implying only the lower bound, and nothing more. I suppose shaking these out (and committing to something about them) is probably a good pre-GA activity.

Reply:
https://github.com/MaterializeInc/materialize/issues/13711